File: bestan-common/TestLua.lua
local map = {
    test = 2112345678,
    nums = { 1, 2, 3 },
    map = {
        [1] = { v1 = 11, v2 = 100,  },
        [20] = { v1 = 100, v2 = 1000, },
    },
    lists = {
        {v1 = 11, v2 = 22},
        {v1 = 111, v2 = 222},
    },
    string = "123123",
    dmap = {
        [1] = {
            [10] = 11,
        },
    },
    dList = {
        { [10]={
            [1] = {
                [10] = 11,
                },
            },
        },
    },
    number = "aaa",
    lvalue = 9007199254740991,
}
print("aaaa")

return map;

File: bestan-common/src/test/java/bestan/common/lua/TestLua.java
package bestan.common.lua;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import bestan.common.lua.LuaParamAnnotation.LuaParamPolicy;

@LuaAnnotation(optional=true)
public class TestLua extends BaseLuaConfig {
	public int test = 11;
	public List<Integer> nums = new ArrayList<Integer>();
	public Map<Integer, ValueLua> map = new HashMap<>();
	public List<ValueLua> lists = new ArrayList<>();
	public String string = "aaa";
	public Map<Integer, Map<Integer, Integer>> dmap = new HashMap<>();
	public List<Map<Integer, Map<Integer, Map<Integer, Integer>>>> dList = new ArrayList<>();
	@LuaParamAnnotation(policy=LuaParamPolicy.OPTIONAL)
	public long lvalue = 122313L;
	
	public static class ValueLua extends BaseLuaConfig{
		public Integer v1 = 0;
		public Integer v2 = 0;
	}
}
